### PR TITLE
Use auto diff to get at jacobian of deflection angles

### PR DESCRIPTION
**Pull request by CKrawczyk**
These changes add the ability to use JAX to find the jacobian of deflection angles.  To do this it first needs a version of the deflection angle function that takes in two scalars and returns a 2 element vector.  From this `jax.jacfwd` is used to get the 2x2 jacobian for a single (y,x) value.  This function is vectorized to take in a set of (y,x) values (of arbitrary shape) and return a (*shape, 2, 2) array of jacobian values.

The final step is to pass the value from a `grid` object into this function.

### CODE REVIEW

File: autogalaxy/operate/deflections.py [hunk 1]
```diff
@@ -709,6 +726,29 @@ def einstein_mass_angular_from(
 
         return einstein_mass_angular_list[0]
 
+    def jacobian_stack(self, y, x, pixel_scales):
+        if not use_jax:
+            return
```
Comment (rhayes777) [inline autogalaxy/operate/deflections.py:731]: Should this ever get called if jax is off?

**Review by rhayes777 — APPROVED** [review]

File: autogalaxy/operate/deflections.py (hunk 1 above)
Comment (CKrawczyk) [reply autogalaxy/operate/deflections.py:731]: I don't expect it to, but figured just to be safe.

Comment (rhayes777) [reply autogalaxy/operate/deflections.py:731]: Might be worth raising an exception instead?

Comment (rhayes777) [reply autogalaxy/operate/deflections.py:731]: Something like: 

```python
if not use_jax:
    raise AssertionError("Function should not be called unless JAX is in use")

return ...

**Pull request merged**